test/core/image_processing/simple_kernels.cpp

```diff
@@ -0,0 +1,143 @@
+#include <boost/gil/image_processing/numeric.hpp>
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/typedefs.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+namespace gil = boost::gil;
+
+bool are_equal(gil::gray32f_view_t expected, gil::gray32f_view_t actual)
+{
+    if (expected.dimensions() != actual.dimensions())
+        return false;
+
+    for (long int y = 0; y < expected.height(); ++y)
+    {
+        for (long int x = 0; x < expected.width(); ++x)
+        {
+            if (expected(x, y) != actual(x, y))
```

Comment:
comparing floating-point numbers with operator== and operator!= is difficult and causes compiler warnings. Can we use an implementation that takes an additional `epsilon` parameter (with a suitable default value) and checks for approximate equality instead ? The internet should have plenty of references...

Reply:
I believe the main problem is that different functions might be used to generate those values, and may be different precision. It seems like deviation from sample values is not constant and has large window of variation.

Reply:
Also Mateusz asked if I could make PR against upstream directly, would you mind if I close this one and create one against upstream? I will do that after fixing the problem.

Reply:
Sure, go ahead.